WordPress/gutenberg
# Add support for @ignore in the docgen tool

## What?
The current `docgen` tool does not recognize the `@ignore` tag. This PR adds that functionality into the tool to allow items to be marked as `@ignore` so that they won't be added to the documentation. 

This is part of a larger effort to improve the Data Modules reference - https://github.com/WordPress/gutenberg/issues/42125

## Why?
Currently, only items tagged as `@private` are not documented and if the tool also supports `@ignore` it will allow us to match how [core indicates that an item should not be documented.](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#phpdoc-tags)

## Testing Instructions
1. Go to the `package/docgen` directory
2. Create a new file called `test.js` in the root with the following code:
```js
/**
 * Returns something from state.
 *
 * @param {Object} state
 * @param {string} data
 *
 * @return {string} The thing we wanted from state.
 */
export function getThingFromState( state, data ) {
	return state[ data ];
}
```
3. Run `node ./bin/cli.js test.js` and check the generated file to confirm the function is documented
4. Add the `@ignore`
5. Run `node ./bin/cli.js test.js` and check the generated file to confirm the function is  NOT documented
